src/cljc/matthiasn/systems_toolbox/component.cljc

```diff
@@ -33,6 +37,16 @@
    :msgs-on-firehose      true
    :reload-cmp            true})
 
+(defn add-to-msg-seq
+  "Function for adding the current component ID to the sequence that the message has traversed
+  thus far. Component IDs can be added either when the message enters or leaves a component.
+  There's a test to avoid conjoining the same ID twice in a row."
+  [msg-meta cmp-id]
+  (let [cmp-seq (vec (:cmp-seq msg-meta))]
```

Comment:
Hadn't seen your second comment when commenting. But then we would miss the origin of the message, which is probably more important than some intermediate steps...